### DISCUSSION TITLE
Initial page for useInfiniteQuery

### DISCUSSION THREAD

**Discussion by mareson** [TanStack/query#5118] [Q&A]
Hello,
I've tried looking in the documentation and here, but I haven't found anything similar to what I'm trying to achieve. I have a table with an infinite scroll and I would like to store the current last page in a url query parameter. When the page is refreshed (or if someone clicks on a link with a query parameter indicating a page), I retrieve the query parameter and would like to force the useInfiniteQuery hook to retrieve all pages that precede that number (i.e. retrieve all content). For example, as a user I will scroll and get to /something/special?page=4. If I do a refresh, I will want to load pages 1, 2, 3 and 4 and display them the same as if the user scrolled to page 4. Thank you very much for your advice.

**Comment by arnaudbzn** [comment]
For performance reasons you may prefer to use a [bidirectional infinite query](https://tanstack.com/query/latest/docs/react/guides/infinite-queries#what-if-i-want-to-implement-a-bi-directional-infinite-list).
If not and if the URL param page number value is 30, all the 29 previous pages will be loaded...
